### ISSUE TITLE
Implement a skill system

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#5]
```
It would be really nice to have some skill system implemented for the players 
to gain advanced features. Every third level or so the player should be able to 
choose a new skill. Some skills should only be available when a certain level 
has been reached.

Ideas for skills to implement:
Increased weapon chance
Increased weapon damage
Better shop prices
Increased dodge chance
Damage resistance
Higher chance of critical hit
Higher multiplier for critical hit
More Action points per turn
More gold dropped from monsters
More exp from killed monsters
Cleave: When a monster has been killed, AP is restored to max
Corpse eater, +1hp for each kill
Berserker, better attack dmg when health <= 20%
Increased fortitude, gain +2hp every level up


Some work on the skill system has already started, but it needs to be 
completed. There are placeholder on the player object to store this 
information, but not a lot more has been completed.
```

Original issue reported on code.google.com by `oskar.wi...@gmail.com` on 20 Nov 2010 at 10:00

**Comment by GoogleCodeExporter** [comment]
Original comment by `oskar.wi...@gmail.com` on 20 Nov 2010 at 10:01

**Comment by GoogleCodeExporter** [comment]
```
We should do some conceptional thinking here:

1) Do we want only separate skills:
a
b
c

2) Or depending skills:
  a   b
 / \  |
c   d e

3) Or skills only available at certain levels:
a level 3+
b
c level 12+

4) Or skills that could be skilled several times and get better each time:
a - better shop prices grade 1: 10% lower prices
b - better shop prices grade 2: 20% lower prices, 15% trade disadvantage 
lovered to 10%
c - better shop prices grade 3: 30% lower prices, 15% trade disadvantage 
lovered to 5%

5) Skills that need a different amount of skill points.
2 skill points to skill: Corpse eater, +1hp for each kill
7 skill points to skill: Cleave: When a monster has been killed, AP is restored 
to max
BTW: I think the described cleave is too good, perhaps should be changed to 
"when fight is over" - but IMO its still too good.

6) Combinations of (1) - (5) are possible
IMO a good way would be combining (4) with (5)
So there are several grades (3-5) of the skill with the need of increasing 
skill points for the level. So in the beginning it is easy to skill the lower 
grades of several skills, later you could skill higher grades. (or you 
specialize on one skill only)

7)
IMO there should be the possibility to reskill the given skill points. Perhaps 
by loosing some money or xp for that or by skilling the reskill-skill ;-)

You did make a good list of skills there. IMO the skill system should be the 
next big thing. (perhaps after 0.6.7) So we should gather skill ideas here. And 
please comment my concept above, because a good concept what we want makes 
implementing far easier.
```

Original comment by `SamuelPl...@gmail.com` on 4 Dec 2010 at 8:13

**Comment by GoogleCodeExporter** [comment]
```
Those are some very good points Samuel! This is a very interesting subject 
about the future gameplay, and also replayability of the game. I agree with you 
that we should start discussing this to find a skill system that works well 
before we start implementing it.

I'll first start with my reflections on your numbered questions:

1,2) My view is that we make a skill tree. An example would be the skill 
"Damage resistance", I think that skill should only possible to select if you 
have reached sufficient level in some prerequisite skill. This way we can make 
the skill tree into something similar to what other games call "classes". A 
warrior type character would preferably choose the warrior skills, and a sneaky 
character would pick the sneaking-, backstabbing- and conversation skills. 
Allowing the player to customize and specialize in one skill tree could 
increase the replayability of the game. ("What would the game be like if I had 
chosen a sneaky character instead?, maybe I should play it again and find out.")

3) Yes, absolutely. Some skills should only be reachable after having reached a 
certain level. Or a combination with (2), so that you can only select a skill 
if you have sufficient skill level in the prerequisite skill. My thought is 
that some skills will be more powerful than other skills, and it could make the 
game unbalanced if the player can choose the powerful skills early in the game. 
("Cleave" as an example).

4) Yes, I think it should be possible to put skill points into a skill several 
times to increase the effect of the skill. This way you could specialise in one 
particular skill if that suits your type of gameplay you want (which in effect 
increases the replayability of the game). Almost all skills that we have 
discussed are possible to divide into several distinct steps, so this should be 
easy to implement. Chance-modifying skills could give a linear increase in 
percentage for each skill point, and so on.

5) No, this would make it confusing if we also do (4) I think. (4) alone could 
mitigate the fact that some skills are already powerful, if we find ways to 
divide the skill effects. "Cleave" as an example could be several steps, one 
skillpoint to gain +1AP, two skillpoint to gain +2AP and so on.

6) Absolutely, some skills should be much harder to gain. If we make it so that 
some skills are only reachable after a certain level, or after having selected 
enough prerequisiste skills, I think we will accomplish this. Keep in mind that 
the amount of exp required to level up is currently exponential, so if we tweak 
the prerequisistes sufficiently, the amount of work required to gain higher 
level skills would also be exponential.

7) Sure, by playing another character :)  I'm sure this is something that will 
be requested a lot from some players after having played a while, when they 
feel that they have chosen the "wrong" skill. I want to make the skill system 
in such a way that you cannot be good at all aspects at the same time, but 
instead have to specialise. Kind of like character classes. If you want, you 
could ofcourse select "all" skills and not specialise, but then you would not 
gain as much advantade as compared to a player that has specialised. This 
specialization is in my mind (as stated above) something that should drive the 
replayability of the game. Allowing the player to redistribute assigned skill 
points would reduce the replayability. That way, you could redistribute all 
skill points to the conversational skills whenever you needed them, and 
redistribute to the warrior skills whenever you get into a fight. That is not 
somthing that we should encourage I think. Either the player chooses the 
warrior path or the player chooses the sneaky path, for example.


A continuation of some thought about what skills to implement:
 - Lesser cleave: Gain n AP after each monster kill
 - Fortune finder: Increased chance of getting better items from monsters
 - Smooth talker: enables certain conversation options
 - Comprehension: +n% exp gained
 - Coward: increased chance of successful fleeing attempts

What other skills could be create that are non-combat related? A lof of the 
skills that we have discussed now are ones that affect combat. I thik we should 
start thinking about skills that affect other gameplay as well. Bartering 
effects, shoplifting?, conversational skills, exp and levelling modifiers, 
intelligent insight?

Thanks again Samuel for bringing this up. The skill system really has the 
potential of making the gameplay very interesting!
```

Original comment by `oskar.wi...@gmail.com` on 11 Dec 2010 at 10:14

**Comment by GoogleCodeExporter** [comment]
```
I like your thoughts about the replayability of the game.

After playing the game for many hours I gained level 20. So I think I could 
reach level 50 perhaps 60 until the game ends. Average should be around 40 to 
50.

So I propose:
 - there are 20 skills listed in this issue - lets assume we have 30+ skills in the end
 - if each skill has 7 grades, then the sum of skill points to spend totally is 210
 - if the player gets 1 skill point each level, then he could only skill around 1/4 of the total available skills

It will be hard to display a skill tree on small android devices.
So I propose a category selection screen:

Weapon skills
  7 / 50

Combat skills
  0 / 40

Defense skills
  4 / 35

Physical skills
  2 / 30

non combat skills
  3 / 40

Other skills
  0 / 15

If I choose a category then skill details for each skill in the category are 
shown similar to the Quest screen.
In collapsed mode only the skill name and its level is shown.
In extended mode there are informations about
 - how does the skill work
 - what is required (hero level, other skills that have to be skilled earlier)
 - what is the current level (for example gain 2 HP for killed monster)
 - what will the next level be (for example gain 3 HP for killed monster)
```

Original comment by `SamuelPl...@gmail.com` on 11 Dec 2010 at 3:26

**Comment by GoogleCodeExporter** [comment]
```
Proposal:
To discuss skills it would be the best to format them like this:

Skill ID:     #1
Skill name:   Hand of Midas
Description:  You gain more gold from monsters
Requirements: none
Grade #1:     20% more gold
Grade #2:     40% more gold
Grade #3:     60% more gold
Grade #4:     80% more gold
Grade #5:     100% more gold
Grade #6:     150% more gold
Grade #7:     200% more gold

Then we could easily refer to them:
Skill #1 needs adjustment in its grades. IMO Grade 7 should only give 100% more 
gold.
Also it would be easy to implement then.
```

Original comment by `SamuelPl...@gmail.com` on 11 Dec 2010 at 3:45

**Comment by GoogleCodeExporter** [comment]
```
I will post more skills if you "confirm" my last 2 posts.
```

Original comment by `SamuelPl...@gmail.com` on 11 Dec 2010 at 3:46

**Comment by GoogleCodeExporter** [comment]
```
Be careful with non-combat skills in a combat game. There needs to be a very 
significant benefit to them to make them worth taking over combat skills in a 
game where you will be fighting from beginning to end. 
Why would you put (using the example above) 7 skill points into getting three 
times as much gold when you could put the skill points into combat skills and 
kill things faster and just kill some extra monsters (and get some extra xp 
this way too.) If it was 700% more gold at Grade 7, maybe. But then can you use 
that extra money to buy potions and equipment, or bribes, that makes up for the 
loss of combat skills? Will the skill still be as useful as the combat skill 
points would be later in the game? It all depends on the value of money to the 
player. In most games money stops being an issue after a while. You either get 
huge amounts of it, or you get all your gear from monster drops which you can 
sell for lots of money. Either that or you always need more money, and that is 
an aspect of real life most people would rather get away from.
Personally, when confronted with a choice like the one above, I will ALWAYS max 
my combat skills first, as that is ultimately what is needed to progress in a 
game, then work on non-combat skills when I have maxed my combat potential. 
After all, what good is a ton of money if you don't have the combat prowess to 
kill the final boss in the game?

It is incredibly tough to balance, and that is just one possible non-combat 
skill. Ultimately, most of them do come back to money though. Stealing gear 
rather than buying it, charming people instead of bribery, better item drops 
rather than having to buy gear. It would be much simpler if there was no money. 
Socialist Andor's Trail! Iron shortsword and leather armour for every citizen!

It may be worth using seperate systems for combat and non-combat skills. 
Seperate experience for soldier and citizen skills. I haven't thought about it 
in depth but it is a possible alternative. I am mostly just spewing ideas 
currently.
```

Original comment by `caveman....@gmail.com` on 12 Dec 2010 at 1:28

**Comment by GoogleCodeExporter** [comment]
```
Personally I'm a big fan of the Fallout-series of games. I once played through 
fallout 2 with a character that was very proficient in pickpocketing, 
explosives, sneak and conversation. I could evade most fights by reverse 
pickpocketing explosives or talking my way into a better position. That type of 
gameplay is something that I find very attractive, that he player has the 
ability to choose the type of gameplay, and has tha possibility of completing 
the game in several possible ways. Maybe it's more of a wish than something we 
can effectively pull off in Andor's Trail. My view is that if we manage to 
balance it correctly, it will dramatically improve the replayability of the 
game, and in effect the overall enjoyment for the player. Maybe it's just my 
subjective judgements, and most people will want to do the warrior-type 
character? That's my long-term vision anyway.

Good ideas caveman, please keep them coming! This is the type of discussions we 
should be having to make the game one of the best around on Android.

@Samuel: Good points regarding the UI of the skill tree. You are absolutely 
right in that we are very limited in what we can display. A grouped view could 
work nice. Do you think we can combine the screen where we view what skills the 
player currently has with the view where the player chooses new skills 
(including skills that the player does not posess yet)?
```

Original comment by `oskar.wi...@gmail.com` on 12 Dec 2010 at 10:08

**Comment by GoogleCodeExporter** [comment]
```
2caveman.android:
I disagree about non combat skills. Gaining more gold per creep would help much 
in andors trail. You could buy better stuff. In the beginning its hard to get 
those golden rings of damage. The skill "Hand of Midas" grade 7 could help much 
about this. And more expensive items are coming.

Each player has its preferences. And if noone picks a skill we should make it 
stronger.
Offering a wide variety of skills will make everyone happy, because he can do 
it in his way.

2oskar.wiksten
I think it will be hard to provide this non combat way to solve the game. Many 
quests must have 2 or more solutions. I really like the idea, but it will be 
hard work.

The idea of a section with "skilled skills" ;-) is a good one.
Could you comment on posting #5? Is this a good way to gather the skills?
Something to add or remove? More or less than 7 grades?
Should I (and others) propose more skill suggestions of that type.
```

Original comment by `SamuelPl...@gmail.com` on 12 Dec 2010 at 2:30

**Comment by GoogleCodeExporter** [comment]
```
Yes, I think the way you propose in comment 5 is a good way to describe the 
skills. Preferably, the effect should be a function of the skill level, so we 
can calculate it dynamically.
```

Original comment by `oskar.wi...@gmail.com` on 12 Dec 2010 at 2:37

**Comment by GoogleCodeExporter** [comment]
```
Dont you think it would look strange for the user to get 133.33% more gold.
It's mathematically always possible to find a function that passes n points. ;-)
```

Original comment by `SamuelPl...@gmail.com` on 12 Dec 2010 at 2:42

**Comment by GoogleCodeExporter** [comment]
```
Regarding comment 8:
I do agree that alternative routes and methods for completing the game would be 
good. Done right, it would add depth, variety and replayability to the game.
Essentially, I think what you want is a puzzle game where one of the possible 
solutions to each quest/dungeon is killing everything, and which of the 
possible solutions you can get to is governed by skills.
I would like to see this stuff in the game, I just can't see how or where it 
can fit into the game as I see it now. That said, it's your game, don't let me 
stop you making it how you want.
```

Original comment by `caveman....@gmail.com` on 13 Dec 2010 at 5:29

**Comment by GoogleCodeExporter** [comment]
```
Added a function to skill #1

Skill ID:     #1
Skill name:   Hand of Midas
Description:  You gain more gold from monsters
Requirements: none
Function:     F(G) = (2/49) * G*G
Grade #1:     4% more gold
Grade #2:     16% more gold
Grade #3:     37% more gold
Grade #4:     65% more gold
Grade #5:     102% more gold
Grade #6:     147% more gold
Grade #7:     200% more gold
```

Original comment by `SamuelPl...@gmail.com` on 13 Dec 2010 at 11:50

**Comment by GoogleCodeExporter** [comment]
```
Skill ID:     #2
Skill name:   Wisdom
Description:  You gain more xp from monsters
Requirements: none
Function:     F(G) = (2/49) * G*G
Grade #1:     4% more xp
Grade #2:     16% more xp
Grade #3:     37% more xp
Grade #4:     65% more xp
Grade #5:     102% more xp
Grade #6:     147% more xp
Grade #7:     200% more xp
```

Original comment by `SamuelPl...@gmail.com` on 13 Dec 2010 at 11:55

**Comment by GoogleCodeExporter** [comment]
```
Skill ID:     #3
Skill name:   Critical Hit
Description:  You increase your chances to do a critical hit.
Requirements: none
Function:     -
Grade #1:     Multiplier 1.5 - Chance 10%
Grade #2:     Multiplier 2 - Chance 15%
Grade #3:     Multiplier 2 - Chance 25%
Grade #4:     Multiplier 3 - Chance 25%
Grade #5:     Multiplier 3 - Chance 35%
Grade #6:     Multiplier 4 - Chance 35%
Grade #7:     Multiplier 4 - Chance 45%

If a weapon gives Critical hit then the overall multiplier / chance should be 
calculated like this:
M_overall = max(M_skill, M_weapon)
C_overall = C_skill + (1 - C_skill) * C_weapon

So far my suggestion.
```

Original comment by `SamuelPl...@gmail.com` on 13 Dec 2010 at 12:13

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
OK, heres a tree of skills that I've been thinking on. Basicly, if skilled 
fully it could make players into full berserkers, but the tree would require 
lots of skillpoints to unlock the final skill and counteract some of the 
penalties.
(I'll put the Skill ID on the format a.b where a is the post number and b is 
skill number for easier reference to multiple permutations of the same skill)

Skill ID:     #16.4
Skill name:   Violent Outburst
Type:         Toggle on/off
Description:  While active, player gets a small bonus to damage and a penalty 
to hit% and dodge%
Requirements: None
Function:     No idea.
Grade #1:     10% damage bonus, 5% penalties
Grade #2:     15% damage bonus, 10% penalties
Grade #3:     20% damage bonus, 15% penalties
Grade #4:     25% damage bonus, 20% penalties

Skill ID:     #16.5
Skill name:   Berserk
Type:         Passive. Always on.
Description:  Player gets a bonus to damage that increases as hp falls, and a 
penalty to hit% and/or dodge% that increases as hp falls.
Requirements: 16.4
Function:     See below.

Skill ID:     #16.6
Skill name:   Battle Frenzy
Type:         Passive. Always on.
Description:  Player gets a bonus to crit% that increases as hp falls, and a 
penalty to hit% and/or dodge% that increases as hp falls.
Requirements: 16.4
Function:     See below.

Skill ID:     #16.7
Skill name:   Rage
Type:         Passive. Always on.
Description:  Player gets a bonus to crit multiplier that increases as hp 
falls, and a penalty to hit% and/or dodge% that increases as hp falls.
Requirements: 16.4
Function:     See below.

Skill ID:     #16.8
Skill name:   Red Mist
Type:         Passive. Always on.
Description:  Player gets a bonus to damage resistance that increases as hp 
falls, and a penalty to hit% and/or dodge% that increases as hp falls.
Requirements: 16.4
Function:     See below.

Skill ID:     #16.9
Skill name:   Clarity
Type:         Passive. Always on.
Description:  Player gets a bonus to hit% and dodge% that partially counteracts 
the penalties from the other skills.
Requirements: 16.5, 16.6, 16.7, 16.8
Function:     See below.

This gives a tree that looks something like:

            4
          / | \
         / / \ \
        5 6   7 8
         \ \ / /
          \ | /
            9

I haven't thought about exact numbers, but 
y=30*(1.03^(-x))
(put it into http://graph-plotter.cours-de-math.eu/ and set both axes to -5 to 
100) where y = bonus and x = percent of player health remaining shows the shape 
of it. I am no good at maths so there is probably a much easier way to 
calculate that, I just fiddled with the formula until the graph looked roughly 
like what I was thinking of.
The actual bonus would need to vary from skill to skill (x30 crit multiplier is 
a bit much, even at 1hp, but x6 probably isn't). Increased levels in each skill 
could either raise the entire line on the graph (increase bonus at all HP), or 
make it flatter (make it a more even progression), or alternate between the two.
the penalties should follow a similar formula, but the penalty from each skill 
should be about 1/3 of the bonus, so the total penalty from all 4 skills is 
larger than the benefit from any one, and the final skill halves the penalty.
```

Original comment by `caveman....@gmail.com` on 13 Dec 2010 at 8:27

**Comment by GoogleCodeExporter** [comment]
```
Skills would certainly be a bonus to the game plot. I also think if there were 
a way to make it a mmorpg. Multi player game with pvp. You would garner a lot 
more game play from a variety of sources, as this would also create more 
revenue. For importance regarding game dynamics, and any upgrades made or the 
costs of maintenance.
```

Original comment by `johnmagu...@gmail.com` on 15 Dec 2010 at 12:52

**Comment by GoogleCodeExporter** [comment]
```
Wow a lot of numbers getting around here. What I could see however is that with 
every single point the benefits are getting more and more serious.
I would turn this thing around.
Say you have 8 skill points per level.
Grades 1,2,3,4,5,6,7 cost 2,3,5,8,13,21,34 skill points (i.e. fibonacci) while 
gains are linear (i.e. 10,20,30,40%). As a final gain, Grade 7 could give some 
extra benefit.
That is, assuming we wish to have so many grades. For me, 3 grades (proficient, 
specialist, master) sounds enough, even though I've grewn up on Diablo II and 
envision so many skills ;)
```

Original comment by `surrano` on 21 Dec 2010 at 7:26

**Comment by GoogleCodeExporter** [comment]
```
I disagree surrano. Oskar wants a system where its important to specialize in 
certain skills. Your system would make me think: I will get every skill at low 
grades. The greatest benefit per skillpoint is at the beginning, so I will not 
specialize at all.

IMO we should give the greatest benefit at the higher grades to the player. So 
he would have fewer but highly graded skills. Would increase replayability of 
the game. And this is one of the things I really would like to see in the game. 
(Skill trees, Quests with alternative endings, ...)
```

Original comment by `SamuelPl...@gmail.com` on 22 Dec 2010 at 10:44

**Comment by GoogleCodeExporter** [comment]
```
a) + 70% AC with sword for 86 points
b) + 50% AC with dagger and +60% AC with sword for 83 points
c) + 40% AC with 5 different weapons for 90 points
d) + 30% AC with 9 different weapons for 90 points

If my character specialises in wielding swords, he'll be willing to sacrifice 
all skill points to get even the slightest improvement. What's the use of being 
proficient with 9 (or even 5) weapons, if you use only 1 kind (maybe 2) on the 
long term? 

This leaves my character with options a) and b).
If we provide some extra (e.g. +20% block or whatever, with any sword) on grade 
7, then this may be the decider. See my note on "final gain".

Of course, you may argue that those are not 9 weapons, but 9 completely 
different skills. Again, see my note on "final gain".

You may even think about synergy, like in Diablo II, or skill trees where 
prerequisites increase for each grade. (e.g. any grade of improved critical 
with sword requires half grade of proficiency with sword)

```

Original comment by `surrano` on 22 Dec 2010 at 12:27

**Comment by GoogleCodeExporter** [comment]
```
what about skills like smithing and fishing? these skills would make the game a 
whole lot more interesting, cant forget about ranged and atk str def, 
atk(attack) for chance of hit, str(strength) for min-max hit, def(defense) 
chance of block and lvls from 1-99, but lower exp till next lvl total 10mil exp 
every skill and tht divided evenly considering every lvl u need more exp 4 a 
lvl up.
```

Original comment by `cristian...@gmail.com` on 24 Dec 2010 at 3:21

**Comment by GoogleCodeExporter** [comment]
```
smithing like creating arms and armour? or customising existing ones?
e.g. improve weapon (+1 damage or +5% attack) needs a "successful skill check" 
(define: skill check?). If the skill check fails, there is a chance that item 
will be broken (crystallised, as in Lineage :) )

fishing like creating food? also needs some drawbacks. Maybe needs to buy 
fishing gear (1) and bait (one per trial), and there is a chance of success. 
Even if it is not successful, there is a chance that fishing gear will be 
broken. Just fishing without risk and selling the food for money is too easy a 
way to earn money... 

Also, for fishing, it would mean that you equip fishing pole instead of deadly 
silver venomous katana of speed, and you may be attacked by wandering monsters 
;)
```

Original comment by `surrano` on 24 Dec 2010 at 9:51

**Comment by GoogleCodeExporter** [comment]
```
well, i was thinking you could actually use those gems you find in stores and 
tht monsters drop as items t2 upgrade existing armour nd for every time you do 
it, you pay a certain amount of coins t2 try nd upgrade

For fishing you would have t2 buy fishing equipment, and type of fish would 
depend on location, like the home town would have the most common fish, and as 
you proceed further the fish would b more rare and worth more, but you have a 
lower chance of catching tht certain fish

Another thing ive been thinking about is the ability t2 cook. This skill would 
go perfectly with the fishing skill and you would b able t2 cook other things 
t2, then chance of success/chance of buring food, dependimg on your skill lvl 
and the food you are cooking.
```

Original comment by `cristian...@gmail.com` on 24 Dec 2010 at 2:45

**Comment by GoogleCodeExporter** [comment]
```
Just reading up a bit on what might be coming in andors trail(finished all of 
the current content).

A couple of thoughts on skills:
some items could possibly have skill  modifiers.
Special skills that could be gained throughout the story or from sidequests.
Skillpoints from doing certain quests or simply adding to a skill directly.
Certain skills locking out others.

Just some food for thought. looking forward to the next update!   
```

Original comment by `wof...@gmail.com` on 3 Jan 2011 at 3:18

**Comment by GoogleCodeExporter** [comment]
```
I dont know if this has been brought up before but if skills are going to be 
implemented i think the monsters will have to be upgraded as well. Either make 
them stronger or give them their own skill sets.
```

Original comment by `bob8...@gmail.com` on 24 Jan 2011 at 9:26

**Comment by GoogleCodeExporter** [comment]
```
Well, I see a lot of really cool ideas, I can see the direction skills are 
going.

There is a lot of potential here. I think that combining the Skillswith the 
class would group things together, and have Jack of All Trades able to learn 
any Skill.

Will you have Masters like in the Breath of Fire series ? Where you apprentice 
and learn skills from them ?

You can leave skills decoupled from classes just fine, have classes be things 
like what improves at level up, or other such bonuses, and have skills be more 
involved.

Are you talking about Skills, groups of information and tasks, or are you 
talking about a technique or Action. I think Actions should be like, Action 
Skills, combat moves, pick lock, disarm trap, etc. And then Passive Skills are 
stuff you just get, like extra dodge or whatever.

Personally, Id like to see most Passive stuff be in classes, and Actions be 
gained from those classes or masters or books/scrolls/artifacts.


```

Original comment by `rqpa...@gmail.com` on 4 Feb 2011 at 3:51

**Comment by GoogleCodeExporter** [comment]
```
Quoting from comment #27:
[Will you have Masters like in the Breath of Fire series ? Where you apprentice 
and learn skills from them ?]

Huh what? That is not breath of fire is it? sounds more like Lufia...

I kind of like the idea of master/apprentice thing. You could be required to 
complete quests in exchange for skill ranks. Although you will get a job/class 
system  instead of "skill system". So I guess it's a good combination to 
satisfy people's "need" for a classic class system...


My suggestions for eventual battle skills to be added (if necessary):

* Coil~passive - Decreases enemy's block chance by x percent -> x = Rank * 2
* Break armor~passive - x% Chance to ignore the enemy's armor for 3-5 turns
* Eagle Eyes~passive(or active?) - Focus on attack chance at the cost block 
chance
* Prayer to Shadow~active - Regenerates Rank * 2 HP per turn for 3 turns.
```

Original comment by `devil...@live.nl` on 1 Mar 2011 at 4:55

**Comment by GoogleCodeExporter** [comment]
```
Work on this has started for v0.6.10.
```

Original comment by `oskar.wi...@gmail.com` on 16 Jul 2011 at 8:54
- Changed state: **Started**
- Added labels: **Milestone-0.6.10**

**Comment by GoogleCodeExporter** [comment]
Original comment by `oskar.wi...@gmail.com` on 16 Oct 2011 at 8:57
- Changed state: **Ready**

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.10 . Closing.
```

Original comment by `oskar.wi...@gmail.com` on 21 Oct 2011 at 8:56
- Changed state: **Fixed**
